backend/de.metas.contracts/src/main/java/de/metas/contracts/modular/workpackage/impl/InterimContractLogsHandler.java

```diff
@@ -116,9 +120,19 @@ public BooleanWithReason doesRecordStateRequireLogCreation(@NonNull final I_C_Fl
 		final String description = TranslatableStrings.adMessage(MSG_ON_COMPLETE_DESCRIPTION, productName, modularContractLogEntry.getQuantity())
 				.translate(Language.getBaseAD_Language());
 
+		final ProductId productId = ProductId.ofRepoId(flatrateTermRecord.getM_Product_ID());
+		final ProductPrice priceActual = Optional.of(flatrateTermRecord)
```

Comment:
using Optional.of() in such case IMHO it's a bit not natural.
IMHO i would prefer to extract this piece of code to a method called 
```
@Nullable PriceActual extractPriceActual(flatRateTerm)
```
and there if would something like 
UomId uomId = UomId.ofRepoIdOrNull(flatrateTerm.getC_UOM_ID)
and then check if uomId != null
(same for currency).

I would do that instead of 
* check if C_UOM_ID > 0
* and in other places using UomIf.ofRepoId
(for the sake of consistency and not assuming that you know things about the internal structure of int ids)

Reply:
added to `IFlatrateBL`